states/huntsman/sleeping.py

```diff
@@ -1,11 +1,15 @@
 def on_enter(event_data):
     """ """
     pocs = event_data.model
-    pocs.next_state = 'ready'
 
-    # If it is dark and safe we shouldn't be in sleeping state
-    if pocs.is_dark() and pocs.is_safe():
-        pocs.say("Weather is good and it is dark. Something must have gone wrong. Stopping loop")
+    if pocs.is_safe() and pocs.should_retry is False:
+        pocs.say("Weather is good and it is dark. Something must have gone wrong. " +
+                 "Stopping loop.")
         pocs.stop_states()
     else:
-        pocs.say("Another successful night!")
+        # Note: Unit will "sleep" before transition until it is safe
+        # to observe again.
+        pocs.next_state = 'ready'
+        pocs.reset_observing_run()
+
+    pocs.say("Another successful night!")
```

Comment:
This doesn't seem like it should be part of this PR.

Reply:
Nope. More failed attempt to fix the tests, based on moving closer to POCS.